server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java

```diff
@@ -575,6 +575,36 @@ public void testAbortNotStartedSnapshotWithoutIO() throws Exception {
         assertThat(getRepositoryData(repoName).getGenId()).isEqualTo(0L);
     }
 
+    @Test
+    public void testStartWithSuccessfulShardSnapshotPendingFinalization() throws Exception {
+        final String masterName = cluster().startMasterOnlyNode();
+        final String dataNode = cluster().startDataOnlyNode();
+        final String repoName = "test-repo";
+        createRepo(repoName, "mock");
+
+        createTableWithRecord("tbl1");
+        startFullSnapshot(repoName, "first-snapshot").get(5, TimeUnit.SECONDS);
+
+        mockRepo(repoName, masterName).setBlockOnWriteIndexFile();
```

Comment:
in the original patch there is a new dedicated [method](https://github.com/elastic/elasticsearch/commit/f70391c6cca592ff60ad2f96431b3df8ef3d8a7e#diff-a08d9f291e2d2bbc64d9bfc486746de6b08001608841986876687fec726d247bR245-R247) for `setBlockOnWriteIndexFile()`. 

not sure it's important but currently our `mockRepo` does `cluster().`**getInstance**`(RepositoriesService.class, nodeName)` whereas the new method in the reference patch does
`internalCluster().`**getMasterNodeInstance**`(RepositoriesService.class)`

Reply:
I intentionally left those out because I think they hide information that's useful to have at the callside. E.g. in this case the `mockRepo(repoName, masterName)` already hints at the master node being used due to the `masterName` argument. in the ES version you need to follow the implementation to figure out which node is used.

(And yes, the two variants do the same thing)

I did the same already for the ConcurrentSnapshotIT test cases